### ISSUE TITLE
LoRa resume from save state gives me identical LoRa

### ISSUE THREAD

**Issue by RockTheCosmos** [kohya-ss/sd-scripts#954]
I am trying to resume training from a save state I created earlier. It is running through the training like normal, but every subsequent LoRa it outputs is producing images identical to the images produced by the LoRa created at the end of the training that corresponds to the save state I am resuming from. It's like it's not undergoing any training and just spitting out the same LoRa every time, even though it acts like it's going through all the training steps as it typically does. I'm using the same settings I used before except I'm resuming from a saved state and I changed the number of epochs and the scheduler number of cycles (I'm using cosine with restarts). Here is the command I am currently using:

accelerate launch --num_cpu_threads_per_process=2 "./sdxl_train_network.py" --enable_bucket --min_bucket_reso=64 --max_bucket_reso=2048 --pretrained_model_name_or_path="D:/SD models/sd_xl_base_1.0.safetensors" --train_data_dir="C:/Users/user/Pictures/LoRa training" --resolution="1344,1792" --output_dir="C:/kohya_ss/outputs" --logging_dir="C:/kohya_ss/outputs/logs" --network_alpha="4" --save_model_as=safetensors --network_module=networks.lora --text_encoder_lr=5e-05 --unet_lr=0.0001 --network_dim=32 --output_name="uraraka_ochaco_v3" --lr_scheduler_num_cycles="2" --scale_weight_norms="1" --no_half_vae --learning_rate="0.0001" --lr_scheduler="cosine_with_restarts" --train_batch_size="1" --max_train_steps="504" --save_every_n_epochs="2" --mixed_precision="bf16" --save_precision="fp16" --seed="1234" --caption_extension=".txt" --cache_latents --cache_latents_to_disk --optimizer_type="AdamW" --max_data_loader_n_workers="4" --max_token_length=150 --resume="D:/uraraka_ochaco_v2-state" --keep_tokens="1" --bucket_reso_steps=32 --save_last_n_steps_state="1" --save_state --shuffle_caption --xformers --bucket_no_upscale --noise_offset=0.0357 --adaptive_noise_scale=0.00357 --log_prefix=xl-loha --sample_sampler=euler_a --sample_prompts="C:/kohya_ss/outputs\sample\prompt.txt" --sample_every_n_epochs="2"

Any ideas on why the LoRa isn't making any progress after resuming from the saved state?

**Comment by MasroorBabar** [comment]
Kinda Same problem, for some reason I can't continue the training more than three hours, and previously I tried to continue the training from save states with same parameters, but the results of new epoch were similar to the first ones, no progress, still waiting for a descriptive answer

**Comment by RockTheCosmos** [comment]
Changing the scheduler fixed the problem, but that's not ideal. I had to change it to cosine and then induce the restarts manually by running through the number of epochs I was using per restart before and save the state each time, then resume from the save state for each restart. Maybe there's something squirrely about using save states with the cosine with restarts scheduler.

**Comment by RockTheCosmos** [comment]
OK, I'll keep that in mind. Thanks for the info.

**Comment by actionless** [comment]
i think that's still a bug - apparently when resuming, total number of steps being updated for the training "session" itself, but not propagated properly to restored state of the scheduler

`diffusers` examples are working that around by re-creating scheduler again (with new settings) and fast-forwarding it to restored step number
